transport/packet.go

```diff
@@ -37,11 +37,24 @@ type UDPEndpoint struct {
 
 var _ PacketEndpoint = (*UDPEndpoint)(nil)
 
-// Connect implements [PacketEndpoint.Connect].
+// Connect implements [PacketEndpoint].Connect.
 func (e UDPEndpoint) Connect(ctx context.Context) (net.Conn, error) {
 	return e.Dialer.DialContext(ctx, "udp", e.Address)
 }
 
+// PacketDialerEndpoint is a [PacketEndpoint] that connects to the given address using the given [PacketDialer].
+type PacketDialerEndpoint struct {
```

Comment:
By the way, this PR made me realize that there's a lot of duplication between Packet and Stream.
After we submit this, I'll create a PR that leverages Generics to clean up some of the code. For example, this works:

```go
type Endpoint[Conn any] interface {
  Connect(ctx context.Context) (Conn, error)
}

type StreamEndpoint Endpoint[StreamConn]
type PacketEndpoint Endpoint[net.Conn]

type Dialer[Conn any] interface {
  Dial(ctx context.Context, address string) (Conn, error)
}

type StreamDialer Dialer[StreamConn]
type PacketDialer Dialer[net.Conn]
```